source/common/unified_memory_allocation/include/uma/memory_pool.h

```diff
@@ -87,6 +87,33 @@ size_t umaPoolMallocUsableSize(uma_memory_pool_handle_t hPool, void *ptr);
 ///
 void umaPoolFree(uma_memory_pool_handle_t hPool, void *ptr);
 
+///
+/// \brief Retrieve string representation of the underlying pool specific
+///        result reported by the the last API that returned
+///        UMA_RESULT_ERROR_POOL_SPECIFIC. Allows for a pool independent way to
+///        return a pool specific result.
+///
+/// \details
+///     - The string returned via the ppMessage is a NULL terminated C style
+///       string.
+///     - The string returned via the ppMessage is thread local.
+///     - The memory in the string returned via the ppMessage is owned by the
+///       adapter.
+///     - The application may call this function from simultaneous threads.
+///     - The implementation of this function should be lock-free.
+///
+/// \return UMA_RESULT_SUCCESS if the result being
+///       reported is to be considered a warning. Any other result code returned
+///       indicates that the adapter specific result is an error
+/// \param hPool specified memory hPool
+/// \param ppMessage [out] pointer to a string containing provider specific
+/// result in string representation. \return UMA_RESULT_SUCCESS if the result
+/// being
+///       reported is to be considered a warning. Any other result code returned
+///       indicates that the adapter specific result is an error
+enum uma_result_t umaPoolGetLastResult(uma_memory_pool_handle_t hPool,
```

Comment:
I wanted to keep the behavior the same as in UR: https://github.com/oneapi-src/unified-runtime/blob/main/include/ur_api.h

Reply:
@igchor : why do we need this if we have already urGetLastResult ?

Reply:
Because we need to implement memory providers and generic memory pools using UMA API. This means that all errors from allocations/free must go through the UMA API. For example, there might be a code like this in the L0 adapter:

```c++
enum uma_result_t ret = umaMemoryProviderAlloc(...);
```

umaMemoryProviderAlloc will call zeMemAlloc* internally but it will need to translate any error to uma_result_t. Since we cannot support all possible error codes in UMA, we need a mechanism like urGetLastResult.

(technically L0 adapter probably won't call umaMemoryProviderAlloc directly, rather it will call umaPoolMalloc which will handle pooling and calls to umaMemoryProviderAlloc but still, the idea behind error propagation is the same).

Reply:
cool, thx @igchor